api/v1/configurationpolicy_types.go

```diff
@@ -211,120 +175,134 @@ const (
-	AggregatePolicyStatus map[string]*ConfigurationPolicyStatus `json:"aggregatePoliciesStatus,omitempty"`
-	ComplianceState       ComplianceState                       `json:"compliant,omitempty"`
-	ClusterName           string                                `json:"clustername,omitempty"`
+	// ObjectDefinition defines required fields, in whole or in part, to be compared with objects on
+	// the cluster.
+	//
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+
+	// RecordDiff specifies whether (and where) to log the diff between the object on the cluster and
+	// the objectDefinition in the policy. Options are "Log" to log the diff in the
+	// config-policy-controller pod and "None" to not log the diff. Defaults to "None".
+	RecordDiff RecordDiff `json:"recordDiff,omitempty"`
 }
 
-// ComplianceMap map to hold CompliancePerClusterStatus objects
-type ComplianceMap map[string]*CompliancePerClusterStatus
+// ConfigurationPolicySpec defines the desired configuration of objects on the cluster along with
+// how the configuration policy should handle when the cluster doesn't match the configuration
+// policy.
+type ConfigurationPolicySpec struct {
+	Severity           Severity           `json:"severity,omitempty"`
+	RemediationAction  RemediationAction  `json:"remediationAction"`
+	EvaluationInterval EvaluationInterval `json:"evaluationInterval,omitempty"`
+	// +kubebuilder:default:=None
+	PruneObjectBehavior PruneObjectBehavior `json:"pruneObjectBehavior,omitempty"`
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:printcolumn:name="Compliance state",type="string",JSONPath=".status.compliant"
+	// NamespaceSelector defines the list of namespaces to include/exclude for objects defined in
+	// spec.object-templates. All selector rules are ANDed. If 'include' is not provided but
+	// 'matchLabels' and/or 'matchExpressions' are, 'include' will behave as if ['*'] were given. If
+	// 'matchExpressions' and 'matchLabels' are both not provided, 'include' must be provided to
+	// retrieve namespaces.
+	NamespaceSelector Target `json:"namespaceSelector,omitempty"`
 
-// ConfigurationPolicy is the Schema for the configurationpolicies API
-type ConfigurationPolicy struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// 'object-templates' is an array of object configurations for the configuration policy to check, create, modify, or
+	// delete objects on the cluster. Keys inside of the objectDefinition in an object template may point to values that
+	// have Go templates. For more advanced Go templating such as `range` loops and `if` conditionals, use
+	// 'object-templates-raw'. Only one of 'object-templates' and 'object-templates-raw' may be set in a configuration
+	// policy.
+	// For more on the Go templates, see https://github.com/stolostron/go-template-utils/blob/main/README.md
```

Comment:
I like this link in upstream but I'm thinking we probably want to avoid it in the ACM version (i.e. Stolostron).

Reply:
That's true. For `stolostron`, we could have a doc link instead (though I'm not a huge fan of keeping up with the version there).